src/Images/ComponentSelector.jsx

```diff
@@ -20,6 +20,32 @@ function ComponentSelector({ service }) {
       state.context.images.actorContext.get(name).image?.imageType.components
   )
 
+  //re-render when click on check-boxes
+  // ideally use shallowEqual as a third entry for useSelector for newer versions
+  const componentVisibilitiesLength = useSelector(
+    service,
+    (state) =>
+      state.context.images.actorContext.get(name).componentVisibilities.length
+  )
+
+  const componentVisibilities = []
+
+  for (let i = 0; i < componentVisibilitiesLength; i++) {
+    componentVisibilities.push(
+      useSelector(
+        service,
+        (state) =>
+          state.context.images.actorContext.get(name).componentVisibilities[i]
+      )
+    )
+  }
```

Comment:
As discussed in person, this is not a good use of React hooks since it includes a hook in a loop. However, changing this to 
```
const componentVisibilities = useSelector(
service,
(state) =>
state.context.images.actorContext.get(name).componentVisibilities,
(a, b) => a.every((value, index) => value===b[index] )
)
```
or something similar does not work. It needs more investigation and I can open an issue about it.